src/Microsoft.DotNet.XHarness.Android/ApkHelper.cs

```diff
@@ -0,0 +1,33 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+
+namespace Microsoft.DotNet.XHarness.Android
+{
+    public static class ApkHelper
+    {
+        public static List<string> GetApkSupportedArchitectures(string apkPath)
+        {
+            if (!File.Exists(apkPath))
+            {
+                throw new FileNotFoundException($"Invalid APK Path: '{apkPath}'");
+            }
+            if (!Path.GetExtension(apkPath).Equals(".apk", StringComparison.OrdinalIgnoreCase))
+            {
+                throw new InvalidOperationException("Only know how to open APK files.");
+            }
+
+            using (ZipArchive archive = ZipFile.Open(apkPath, ZipArchiveMode.Read))
+            {
+                // Enumerate all folders under /lib inside the zip
+                var allLibFolders = archive.Entries.Where(e => e.FullName.StartsWith("lib/"))
```

Comment:
I guess this enumerates files in `lib/` as well? Can that be bad somehow?

Reply:
if you put a file in there there (I didn't find any like this) instead of an arch-named folder, I'd say that was a supported architecture.  I think it's not worth worrying about now;  this is a fallback and ideally the users will tell us what it is before it gets there.